mlir/lib/Conversion/ConvertToLLVM/ToLLVMInterface.cpp

```diff
@@ -30,3 +30,20 @@ void mlir::populateConversionTargetFromOperation(
                                                    patterns);
   });
 }
+
+void mlir::populateOpConvertToLLVMConversionPatterns(
+    Operation *op, ConversionTarget &target, LLVMTypeConverter &typeConverter,
+    RewritePatternSet &patterns) {
+  auto iface = dyn_cast<ConvertToLLVMOpInterface>(op);
+  if (!iface)
+    return;
+  SmallVector<ConvertToLLVMAttrInterface, 12> attrs;
+  iface.getConvertToLLVMConversionAttrs(attrs);
+  for (ConvertToLLVMAttrInterface attr : attrs)
+    attr.populateConvertToLLVMConversionPatterns(target, typeConverter,
+                                                 patterns);
+}
```

Comment:
This seems a bit "ad-hoc" to me, and also it means that running the pass on a a module with such an attribute, or on a function inside the module would give a different result?

Reply:
I think there's a missing `CompilationUnitOpInterface` abstraction in general in MLIR, which would clarify the semantics in this case, because one could search the nearest op with the interface (or trait).

However, I also think it is reasonable for users to expect that a pass might perform different actions depending on the op context. In the sense that `convert-to-llvm` applied to a function won't update `func.func`, but if applied to `module` containing `func.func` it will be converted to `llvm.func`.

Reply:
> However, I also think it is reasonable for users to expect that a pass might perform different actions depending on the op context.

Yes but here it's not the "op context": what I'm describing is the exact same input IR, but the pass pipeline configured differently

Reply:
How about if the anchor interface is changed to:

```c++
ConvertToLLVMOpInterface getConvertToLLVMOpInterface(Operation* op) {
  if (auto iface = dyn_cast<ConvertToLLVMOpInterface>(op))
    return iface;
  return op->getParentOfType<ConvertToLLVMOpInterface>();
}
// ... ConvertToLLVMPass::runOnOperation()
  ConvertToLLVMOpInterface iface = getConvertToLLVMOpInterface(getOperation());
  ...
```

In that case, anything inside of a `ConvertToLLVMOpInterface` would get converted according to a target.